types/field_type.go

```diff
@@ -146,6 +146,37 @@ func hasVariantFieldLength(tp *FieldType) bool {
 	return false
 }
 
+// http://cavaliercoder.com/blog/optimized-abs-for-int64-in-go.html
```

Comment:
It may be better to extract these newly added functions into a new file `types/util.go`.